packages/solana/src/connectors/walletConnectConnector.ts

```diff
@@ -144,16 +146,56 @@ export class WalletConnectConnector extends BaseConnector implements Connector {
     return { signatures: [{ signature: base58.encode(transaction.serialize()) }] }
   }
 
-  public async sendTransaction(transactionParam: Transaction | VersionedTransaction) {
+  private async _sendTransaction(transactionParam: Transaction | VersionedTransaction) {
     const encodedTransaction = (await this.signTransaction(transactionParam)) as {
       signatures: {
         signature: string
       }[]
     }
+    // Const signedTransaction = transactionParam.serialize()
     const signedTransaction = base58.decode(encodedTransaction.signatures[0]?.signature ?? '')
-    await SolStoreUtil.state.connection?.sendRawTransaction(signedTransaction)
+    const txHash = await SolStoreUtil.state.connection?.sendRawTransaction(signedTransaction)
+
+    return {
+      tx: txHash,
+      signature: base58.encode(signedTransaction)
+    }
+  }
+
+  public async sendTransaction(transactionParam: Transaction | VersionedTransaction) {
+    const { signature } = await this._sendTransaction(transactionParam)
+
+    return signature
+  }
 
-    return base58.encode(signedTransaction)
+  public async signAndSendTransaction(
+    transactionParam: Transaction | VersionedTransaction,
+    signers: Signer[]
+  ) {
+    if (transactionParam instanceof VersionedTransaction) {
+      return 'it doesnt work with versioned transaction'
```

Comment:
Should this throw error or return string? How will this be handled on the higher level?

Reply:
Changed to throw Error